### PR TITLE
Move workflow generator and related parts

### PR DESCRIPTION
**Pull request by milesgranger**
Problem :writing_hand: 
---
Gordo components isn't fully complete without the workflow generator and related parts of the 'workflow' process.

Solution :keyboard: 
---
This creates a `workflow` component which includes these pieces. Only some minor improvements have been made to make this inline with gordo-components. ie. Not using `argparse` but `click`, and therefore using `click.testing.CliRunner` instead of the ugly capturing of stdout use in infrastructure

Some reorganization may be required later, or in this PR?

### CODE REVIEW

File: Dockerfile-argo [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Installs workflow-generator, kubectl and argo. Runs workflow-generator on the machine-config
```
Comment (epa095) [inline Dockerfile-argo]: maybe use the opportunity to rename this to follow the same scheme as the other dockerfiles?

File: pytest.ini [hunk 2]
```diff
@@ -7,4 +7,4 @@ flakes-ignore =
 filterwarnings =
     ignore::DeprecationWarning
     ignore::PendingDeprecationWarning
-timeout = 100
+timeout = 300
```
Comment (epa095) [inline pytest.ini:10]: On purpose ?

Comment (milesgranger) [reply pytest.ini:10]: You betcha. The linting of the generated workflow can take quite some time occasionally. Was at 200 for part of the PR, but think it was just barely over that.

**Review by epa095 — APPROVED** [review]
LGTM

**Pull request merged**